akd/src/append_only_zks.rs

```diff
@@ -199,88 +206,108 @@ impl Azks {
         Ok(load_count)
     }
 
-    /// An azks is built both by the [crate::directory::Directory] and the auditor.
-    /// However, both constructions have very minor differences, and the append_only_usage
-    /// bool keeps track of this.
-    pub async fn batch_insert_leaves_helper<S: Database + Sync + Send>(
-        &mut self,
+    /// Inserts a batch of leaves recursively from a given node label.
+    #[async_recursion]
+    pub async fn recursive_batch_insert_leaves<S: Database + Sync + Send>(
         storage: &StorageManager<S>,
+        node_label: Option<NodeLabel>,
         insertion_set: Vec<Node>,
```

Comment:
nit: can we make this take a `&[Node]` instead of a Vec? That way we don't need to create vectors eveywhere and probably can help cut-down on cloning.

Reply:
Ah I think it actually makes sense for the `insertion_set` to be a Vec here. We're handing over ownership of the partitions to the recursive calls, so we're never cloning the data. If we were to make it a reference, we would have to ensure that the original Vec lives long enough until the recursive call returns, which is particularly inconvenient if the recursive call runs in a separate task.